pkg/apis/config/allowed_types.go

```diff
@@ -13,10 +14,12 @@ var Decoder runtime.Decoder
 // TODO(dibyom): We should have a way of configuring this instead of an init function?
 func init() {
 	scheme := runtime.NewScheme()
+	utilruntime.Must(customrunsv1alpha1.AddToScheme(scheme))
 	utilruntime.Must(pipelineresourcev1alpha1.AddToScheme(scheme))
 	utilruntime.Must(pipelinev1beta1.AddToScheme(scheme))
 	codec := serializer.NewCodecFactory(scheme)
 	Decoder = codec.UniversalDecoder(
+		customrunsv1alpha1.SchemeGroupVersion,
```

Comment:
Ahh I see what happened. This is because the [`customrunsv1alpha1.SchemeGroupVersion`](https://github.com/tektoncd/pipeline/blob/e4085bcd51692f2e59ad77de53a56d6747793796/pkg/apis/pipeline/v1alpha1/register.go#L27) actually equals to [`pipelineresourcev1alpha1.SchemeGroupVersion`](https://github.com/tektoncd/pipeline/blob/e4085bcd51692f2e59ad77de53a56d6747793796/pkg/apis/resource/v1alpha1/register.go#L27)

Reply:
Therefore we only need two schemaGroupVersions here: `pipelinev1alpha1` and `pipelinev1beta1`

Reply:
Thank you @XinruZhang